HUBpayFileUploadGUI.ps1

```diff
@@ -41,20 +71,24 @@ $button.Location = New-Object System.Drawing.Point $buttonLocationX, $buttonLoca
 $button.Size = New-Object System.Drawing.Size(100, 30)
 $button.Text = "Execute Script"
 $button.Add_Click({
-    $albacs = $textboxes["Albacs"].Text
-    $s3bucket = $textboxes["S3 Bucket"].Text
-    $awss3user = $textboxes["AWS S3 User"].Text
-    $loglocation = $textboxes["Log Location"].Text
-    $sauser = $textboxes["SA User"].Text
-    $sapass = $textboxes["SA Password"].Text
-    $TaskName = $textboxes["Task Name"].Text
-    $PASS_VMName = $textboxes["PASS VM Name"].Text
-
-    # Call the script with the provided parameters
-    & "C:\HubPay\s3-scripts\HUBpay-createschedtask.ps1" -albacs $albacs -s3bucket $s3bucket -awss3user $awss3user -loglocation $loglocation -sauser $sauser -sapass $sapass -TaskName $TaskName -PASS_VMName $PASS_VMName
-
-    # Run the create-task.ps1 script
-    Start-Process powershell.exe -ArgumentList "-File '$location\HUBpay-createschedtask.ps1' -albacs '$albacs' -s3bucket '$s3bucket' -awss3user '$awss3user' -loglocation '$loglocation' -sauser '$sauser' -sapass '$sapass' -TaskName '$TaskName' -PASS_VMName '$PASS_VMName'"
+    try {
+        # Retrieve values from textboxes and execute the script
+        $albacs, $s3bucket, $awss3user, $loglocation, $sauser, $sapass, $TaskName, $PASS_VMName = $labels.ForEach({ $textboxes[$_].Text })
+
+        # Docs on this $PSScriptRoot var: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_automatic_variables?view=powershell-7.4#psscriptroot
+        & "$PSScriptRoot\HUBpay-createschedtask.ps1" -albacs $albacs -s3bucket $s3bucket -awss3user $awss3user -loglocation $loglocation -sauser $sauser -sapass $sapass -TaskName $TaskName -PASS_VMName $PASS_VMName
```

Comment:
Double check this change is equivalent

Reply:
Textbox parsing was condensed to one line, in original version `HUBpay-createschedtask.ps1` script was being executed twice (once in the same process using `&`, and again in a separate process using `Start-Process`). Kept it within the same process for now, because it makes error catching easier.

Reply:
Tested this 👍